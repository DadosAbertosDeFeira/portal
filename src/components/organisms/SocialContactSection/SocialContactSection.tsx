import type { SocialMediaType } from "@utils/social";
import { lookupSocialMediaIcon } from "@utils/social";
import { Box } from "atoms/Box";
import { IconButton } from "atoms/IconButton";
import React from "react";

const medias = {
  facebook: process.env.NEXT_PUBLIC_FACEBOOK_URL ?? "",
  github: process.env.NEXT_PUBLIC_GITHUB_URL ?? "",
  instagram: process.env.NEXT_PUBLIC_INSTAGRAM_URL ?? "",
  twitter: process.env.NEXT_PUBLIC_TWITTER_URL ?? "",
};

export function SocialContactSection() {
  return (
    <Box
      as="section"
      className="mt-md grid auto-cols-auto grid-flow-col grid-cols-2 grid-rows-2 gap-md self-center tablet:self-start"
    >
      {Object.entries(medias).map(([key, value]) => {
        return (
          <IconButton
            key={key}
            className="pa-md flex h-xl w-4xl items-center justify-center space-x-sm rounded bg-blue-100 px-sm capitalize"
            as="link"
            href={value}
            icon={lookupSocialMediaIcon(key as SocialMediaType)}
            label={key}
          />
        );
      })}
    </Box>
  );
}
