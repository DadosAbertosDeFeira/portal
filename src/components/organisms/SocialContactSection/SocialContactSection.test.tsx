import { render } from "@testing-library/react";

import { SocialContactSection } from "./SocialContactSection";

describe("SocialContactSection", () => {
  const makeSut = () => {
    return render(<SocialContactSection />);
  };

  it("renders correctly", () => {
    const { asFragment } = makeSut();

    expect(asFragment).toMatchSnapshot();
  });
});
