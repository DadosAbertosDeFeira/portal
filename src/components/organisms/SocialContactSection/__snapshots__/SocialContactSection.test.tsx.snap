// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialContactSection renders correctly 1`] = `[Function]`;
